boa/src/object/internal_methods/arguments.rs

```diff
@@ -0,0 +1,261 @@
+
+                // 2. Assert: setStatus is true because formal parameters mapped by argument
+                // objects are always writable.
+                debug_assert!(set_status)
+            }
+
+            // ii. If Desc.[[Writable]] is present and its value is false, then
+            if let Some(false) = desc.writable() {
+                // 1. Call map.[[Delete]](P).
+                map.__delete__(&key, context)?;
+            }
+        }
+    }
+
+    // 8. Return true.
+    Ok(true)
+}
+
+/// [[Get]] for arguments exotic objects.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-arguments-exotic-objects-get-p-receiver
+pub(crate) fn arguments_exotic_get(
+    obj: &JsObject,
+    key: &PropertyKey,
+    receiver: JsValue,
+    context: &mut Context,
+) -> JsResult<JsValue> {
+    // 1. Let map be args.[[ParameterMap]].
+    let map = obj
+        .borrow()
+        .as_mapped_arguments()
+        .expect("arguments exotic method must only be callable from arguments objects")
+        .parameter_map();
+
+    // 2. Let isMapped be ! HasOwnProperty(map, P).
+    // 4. Else,
+    if map
+        .has_own_property(key.clone(), context)
+        .expect("HasOwnProperty must not fail per the spec")
+    {
+        // a. Assert: map contains a formal parameter mapping for P.
+        // b. Return Get(map, P).
+        map.get(key.clone(), context)
+
+    // 3. If isMapped is false, then
+    } else {
+        // a. Return ? OrdinaryGet(args, P, Receiver).
+        super::ordinary_get(obj, key, receiver, context)
+    }
+}
+
+/// [[Set]] for arguments exotic objects.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-arguments-exotic-objects-set-p-v-receiver
+pub(crate) fn arguments_exotic_set(
+    obj: &JsObject,
+    key: PropertyKey,
+    value: JsValue,
+    receiver: JsValue,
+    context: &mut Context,
+) -> JsResult<bool> {
+    // 1. If SameValue(args, Receiver) is false, then
+    // a. Let isMapped be false.
+    // 2. Else,
+    if JsValue::same_value(&obj.clone().into(), &receiver) {
+        // a. Let map be args.[[ParameterMap]].
+        let map = obj
+            .borrow()
+            .as_mapped_arguments()
+            .expect("arguments exotic method must only be callable from arguments objects")
+            .parameter_map();
+
+        // b. Let isMapped be ! HasOwnProperty(map, P).
+        // 3. If isMapped is true, then
+        if map
+            .has_own_property(key.clone(), context)
+            .expect("HasOwnProperty must not fail per the spec")
+        {
+            // a. Let setStatus be Set(map, P, V, false).
+            let set_status = map.set(key.clone(), value.clone(), false, context)?;
+
+            // b. Assert: setStatus is true because formal parameters mapped by argument objects are always writable.
+            debug_assert!(set_status);
```

Comment:
The spec does not expect `set` to fail here. So we should either `expect` on `set` or assert that `set_status` is `Ok(true)`.